test/common/stream_info/stream_info_impl_test.cc

```diff
@@ -227,29 +232,88 @@ TEST_F(StreamInfoImplTest, MiscSettersAndGetters) {
 TEST_F(StreamInfoImplTest, SetFrom) {
   StreamInfoImpl s1(Http::Protocol::Http2, test_time_.timeSystem(), nullptr);
 
+  // For testing setFromForRecreateStream
   s1.addBytesReceived(1);
   s1.downstreamTiming().onLastDownstreamRxByteReceived(test_time_.timeSystem());
 
+  // For testing setFrom
+  s1.setRouteName("foo");
+  s1.setVirtualClusterName(absl::optional<std::string>("bar"));
+  s1.setResponseCode(200);
+  s1.setResponseCodeDetails("OK");
+  s1.setConnectionTerminationDetails("baz");
+  s1.setUpstreamInfo(std::make_shared<UpstreamInfoImpl>());
+  s1.upstreamInfo()->upstreamTiming().onLastUpstreamTxByteSent(test_time_.timeSystem());
+  s1.onRequestComplete();
+  s1.setResponseFlag(FailedLocalHealthCheck);
+  s1.healthCheck(true);
+  s1.route_ = std::make_shared<NiceMock<Router::MockRoute>>();
+  s1.setDynamicMetadata("com.test", MessageUtil::keyValueStruct("test_key", "test_value"));
+  s1.filterState()->setData("test", std::make_unique<TestIntAccessor>(1),
+                            FilterState::StateType::ReadOnly, FilterState::LifeSpan::FilterChain);
+  Http::TestRequestHeaderMapImpl headers1;
+  s1.setRequestHeaders(headers1);
+  Upstream::ClusterInfoConstSharedPtr cluster_info(new NiceMock<Upstream::MockClusterInfo>());
+  s1.setUpstreamClusterInfo(cluster_info);
+  s1.setStreamIdProvider(
+      std::make_shared<StreamIdProviderImpl>("a121e9e1-feae-4136-9e0e-6fac343d56c9"));
+  s1.setTraceReason(Tracing::Reason::ClientForced);
+  s1.setFilterChainName("foobar");
+  s1.setAttemptCount(5);
+
 #ifdef __clang__
 #if defined(__linux__)
 #if defined(__has_feature) && !(__has_feature(thread_sanitizer))
-  ASSERT_TRUE(sizeof(s1) == 784 || sizeof(s1) == 800 || sizeof(s1) == 808 || sizeof(s1) == 824)
+  ASSERT_TRUE(sizeof(s1) == 800 || sizeof(s1) == 816 || sizeof(s1) == 840)
       << "If adding fields to StreamInfoImpl, please check to see if you "
-         "need to add them to setFromForRecreateStream! Current size "
+         "need to add them to setFromForRecreateStream or setFrom! Current size "
       << sizeof(s1);
 #endif
 #endif
 #endif
 
   StreamInfoImpl s2(Http::Protocol::Http11, test_time_.timeSystem(), nullptr);
-  s2.setFromForRecreateStream(s1);
+  Http::TestRequestHeaderMapImpl headers2;
+  s2.setFrom(s1, &headers2);
```

Comment:
Instead of changing the existing test to test a new interface, can you add another unit test that test the extra thing setFrom() does?

Reply:
Done.